doc/developer-guide/cache-architecture/cache-initialization.en.rst

```diff
@@ -0,0 +1,156 @@
+   to you under the Apache License, Version 2.0 (the
+   "License"); you may not use this file except in compliance
+   with the License.  You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing,
+   software distributed under the License is distributed on an
+   "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+   KIND, either express or implied.  See the License for the
+   specific language governing permissions and limitations
+   under the License.
+
+.. include:: ../../common.defs
+.. default-domain:: cpp
+.. _cache-initialization:
+
+Cache Initialization
+********************
+
+Initialization starts with an instance of :cpp:class:`Store` reading the storage
+configuration file, by default :file:`storage.config`. For each valid element in
+the file an instance of :cpp:class:`Span` is created. These are of basically
+four types:
+
+* File
+
+* Directory
+
+* Disk
+
+* Raw device
+
+After creating all the :cpp:class:`Span` instances, they are grouped by device
+ID to internal linked lists attached to the :cpp:member:`Store::disk`
+array[#store-disk-array]_. Spans that refer to the same directory, disk, or raw
+device are coalesced in to a single span. Spans that refer to the same file
+with overlapping offsets are also coalesced [#coalesced-spans]_. This is all done in
+:func:`ink_cache_init` called during startup.
+
+.. note::
+
+   The span logic is also used by the HostDB and more than one otherwise
+   inexplicable feature is provided by the span logic for that module.
+
+After configuration initialization, the cache processor is started by calling
+:cpp:member:`CacheProcessor::start()`. This does a number of things:
+
+For each valid span, an instance of :cpp:class:`CacheDisk` is created. This
+class is a :term:`continuation` and so can be used to perform potentially
+blocking operations on the span. The primary use of these is to be passed to
+the AIO threads as the callback when an I/O operation completes. These are then
+dispatched to AIO threads to perform :term:`storage unit` initialization. After
+all of those have completed, the resulting storage is distributed across the
+:term:`volumes <cache volume>` in :func:`cplist_reconfigure`. The
+:cpp:class:`CacheVol` instances are created at this time.
+
+Stripe Assignment
+=================
+
+:term:`Cache stripe <cache stripe>` assignment setup is done once all stripes have initialized (that
+is, all stripe header read operations have completed). There is an instance of
+:cpp:class:`CacheHostRecord` for each line in :file:`hosting.config` and one generic instance
+(:member:`Cache::hosttable`) for all cache volumes that are not explicitly assigned. If
+:file:`hosting.config` is empty then all cache volumes will be in the generic record.
+
+:func:`build_vol_hash_table` in :ts:git:`iocore/cache/Cache.cc` does the work of setting up the
```

Comment:
The documentation describing content retrieval from the cache can be linked here.